DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/fixtures/alias/StaticFieldAlias.java

```diff
@@ -0,0 +1,92 @@
+    @NoAlias(reason = "no alias with field and return value", id = 3)
+    @NoAliasLine(reason = "no alias with field and unrelated uVar", lineNumber = 56, methodName = "noAlias")
+    @MayAlias(reason = "may alias with field and return value", id = 2)
+    @MayAliasLine(reason = "may alias with field and returned uVar", lineNumber = 67, methodName = "returnMayAliasField")
+    @MayAlias(reason = "may alias with field and return value via parameter", id = 5)
+    @MayAliasLine(reason = "may alias with field and returned uVar via parameter", lineNumber = 82, methodName = "parameterIsMayAliasField")
+    @MayAlias(reason = "may alias with field and parameter", id = 6)
+    public static Object mayAliasField = new Object();
+
+    @MayAliasLine(reason = "may alias with field and assigned uVar", lineNumber = 61, methodName = "reassignNullField")
+    public static Object nullField = null;
+
+    @MayAlias(reason = "may alias with static final field and return value", id = 4)
+    @MayAliasLine(reason = "may alias with static final field and returned uVar", lineNumber = 73, methodName = "returnStaticFinalField")
+    @MayAlias(reason = "may alias with static final field and return value via parameter", id = 7)
+    @MayAliasLine(reason = "may alias with static final field and returned uVar via parameter", lineNumber = 89, methodName = "parameterIsStaticFinalField")
+    @MayAlias(reason = "may alias with static final field and parameter", id = 8)
+    public static Object staticFinalField = new Object();
+
+    public static void main(String[] args) {
+        noAlias(new Object());
+        noAlias(staticFinalField);
+
+        parameterIsMayAliasField(mayAliasField);
+        parameterIsStaticFinalField(staticFinalField);
+    }
+
+    public static void reassignField() {
+        Object o = new Object();
+        mayAliasField = o;
+    }
+
+    @NoAlias(reason = "no alias with field and return value", id = 3)
+    public static Object noAlias(
+            @NoAlias(reason = "no alias with field and parameter", id = 1)
+            Object a) {
+        Object o = new Object();
+
+        if (Math.random() > 0.5) {
+            o = a;
+        }
+
+        return o;
+    }
+
+    public static void reassignNullField() {
+        Object o = new Object();
+        nullField = o;
+    }
+
+    @MayAlias(reason = "may alias with field and return value", id = 2)
+    public static Object returnMayAliasField() {
+        Object o = mayAliasField;
+        return o;
+    }
+
+    @MayAlias(reason = "may alias with static final field and return value", id = 4)
+    public static Object returnStaticFinalField() {
+        Object o = staticFinalField;
+        return o;
+    }
+
+    @MayAlias(reason = "may alias with field and return value via parameter", id = 5)
+    @MayAliasLine(reason = "may alias with return value and uVar via parameter and field", lineNumber = 43, methodName = "reassignField")
```

Comment:
What is the reason to have @MayAliasLine in addition to @MayAlias? Shouldn't one suffice?

Reply:
They are referencing two different entities. Line 76 states that the return value of the method is in a may alias relation with the field in general. Line 77 states that the return value of the method is in a may alias relation with the uVar that the field is assigned to in line 43.

Reply:
I think my question was more: Why do we need two different types of annotations. Can't one type of annotation express both of these entities?

Reply:
Oh, i see. Currently, the original MayAlias annotation is used if both element can be annotated, e.g., an alias between two parameters. MayAliasLine has been added to explicitly handle cases where at least one of the elements cannot be annotated (is an uVar). Technically, it should be possible to add the `int id();` method to the mayAliasLine annotation with a default value and choosing the resolution strategy (searching for annotation with the same id vs. searching for uVar(s) in given lineNumber) depending on whether id is explicitly set or not. That would eliminate the need for two types of annotations. I will look into that

Reply:
There is now only one type of annotation